### PR TITLE
USHIFT-3177: Avoid updating default router ips if status is the same

### PR DESCRIPTION
**Pull request by pacevedom**
Updates to the network in the host trigger a possible update. This also triggers a log about an update that may be unnecessary (both the log and the update).
By comparing the current status and the new one we can avoid an API call and also log properly.

**Which issue(s) this PR addresses**:

Closes #<Issue Number>

### CODE REVIEW

File: pkg/loadbalancerservice/controller.go [hunk 1]
```diff
@@ -235,13 +236,20 @@ func (c *LoadbalancerServiceController) updateDefaultRouterServiceStatus(ips []s
...
 		}
-		klog.Infof("Updating default router service status: %v", ips)
 		newStatus := &corev1.LoadBalancerStatus{}
 		for _, ip := range ips {
 			newStatus.Ingress = append(newStatus.Ingress, corev1.LoadBalancerIngress{
 				IP: ip,
 			})
 		}
+
+		equal := slices.EqualFunc(svc.Status.LoadBalancer.Ingress, newStatus.Ingress, func(oldIP, newIP corev1.LoadBalancerIngress) bool {
```
Comment (pmtk) [inline pkg/loadbalancerservice/controller.go:246]: Should we sort the slices before comparing? Are slices with the same elements considered equal or order makes a difference?

Comment (pacevedom) [reply pkg/loadbalancerservice/controller.go:246]: They are already sorted here: https://github.com/openshift/microshift/blob/main/pkg/loadbalancerservice/router.go#L119
This was done in a previous PR.

**Pull request merged**